LabBuilder/LabBuilder.psm1

```diff
@@ -4762,6 +4880,200 @@ Function Uninstall-Lab {
+		 #This will have to change depending on the version of Convert-WindowsImage being used. 
+				If ($VMTemplateDisk.Generation -eq $Null) {$VhdPartitionStyle = 'UEFI'}
+				Else {
+				switch ($VMTemplateDisk.Generation)
+					{
+						'1' {[String]$VHDPartitionStyle = 'BIOS'}
+						'2' {[String]$VHDPartitionStyle = 'UEFI'}
+					}
+				} #End Else
+
+				Write-Verbose "VHDPartition Style = $VhdPartitionStyle"
+				
+				$Edition = $VMTemplateDisk.WimImage
+				
+				Write-Verbose "Image to use = $Edition"
+
+				$Sourcepath = "$([string]$DriveLetter):\" + $WimPath
+				Write-Verbose "Source path for convert-windowsimage is $Sourcepath"
+
+				$VHDpath = "$([string]$TemplateDiskPath)\" + $VMTemplateDiskName
+
+				Convert-WindowsImage `
+					-Sourcepath $Sourcepath `
+					-VHDpath $VHDpath `
+					–VHDFormat $VHDFormat `
+					-Edition $Edition `
+					-DiskLayout $VHDPartitionStyle 
+					      
+
+				if ($VMTemplateDisk.IsNano -eq 'True')
+				{
+					$Packages = $TemplateDisk.Packages
+						If (-not (Test-Path -Path $MountFolder -PathType Container)) 
+							{
+								$null = New-Item -Path $MountFolder -ItemType Directory
+							}
+					# Mount the VHD to load packages into it
+					& "$DismFolder\Dism.exe" '/Mount-Image' "/ImageFile:$PathtoDisk" '/Index:1' "/MountDir:$MountFolder"
+
+					$PackageList = @(
+						@{ Name = 'Compute'; Filename = 'Microsoft-NanoServer-Compute-Package.cab' },
+						@{ Name = 'OEM-Drivers'; Filename = 'Microsoft-NanoServer-OEM-Drivers-Package.cab' },
+						@{ Name = 'Storage'; Filename = 'Microsoft-NanoServer-Storage-Package.cab' },
+						@{ Name = 'FailoverCluster'; Filename = 'Microsoft-NanoServer-FailoverCluster-Package.cab' },
+						@{ Name = 'ReverseForwarders'; Filename = 'Microsoft-OneCore-ReverseForwarders-Package.cab' },
+						@{ Name = 'Guest'; Filename = 'Microsoft-NanoServer-Guest-Package.cab' },
+						@{ Name = 'Containers'; Filename = 'Microsoft-NanoServer-Containers-Package.cab' },
+						@{ Name = 'Defender'; Filename = 'Microsoft-NanoServer-Defender-Package.cab' },
+						@{ Name = 'DCB'; Filename = 'Microsoft-NanoServer-DCB-Package.cab' },
+						@{ Name = 'DNS'; Filename = 'Microsoft-NanoServer-DNS-Package.cab' },
+						@{ Name = 'DSC'; Filename = 'Microsoft-NanoServer-DSC-Package.cab' },
+						@{ Name = 'IIS'; Filename = 'Microsoft-NanoServer-IIS-Package.cab' },
+						@{ Name = 'NPDS'; Filename = 'Microsoft-NanoServer-NPDS-Package.cab' },
+						@{ Name = 'SCVMM'; Filename = 'Microsoft-Windows-Server-SCVMM-Package.cab' },
+						@{ Name = 'SCVMM-Compute'; Filename = 'Microsoft-Windows-Server-SCVMM-Compute-Package.cab' }
+					)
+
+					# Add the selected packages
+					foreach ($Package in $PackageList) {
+						If ($Package.Name -in $Packages) {
+							Write-Verbose -Message "Adding Package $($Package.Filename) to Image"
+							& "$DismFolder\Dism.exe" '/Add-Package' "/PackagePath:$($DriveLetter):\NanoServer\packages\$($Package.Filename)" "/Image:$MountFolder"
```

Comment:
At some point I might look at changing this code to use the DISM powershell module, but we'll leave that for another day.

A bit of history: The reason why I used DISM.exe rather than the DISM powershell module was that DISM that came with Win 2K12 R2 didn't work with Nano on Win 2K16 TP2. But I think DISM has been updated on Win 2K12 R2 so that it now works. I'll raise this as an enhancement once this PR is through.